app/controllers/school/devices_controller.rb

```diff
@@ -10,6 +10,8 @@ def order
       end
     elsif @school.can_order_for_specific_circumstances? && @school.can_order_devices?
       render :can_order_for_specific_circumstances
+    elsif @school.can_order?
```

Comment:
I'm not able to follow the logic in this `if ... else ...` block without looking at the template names.

`@school.can_order?` and `@school.can_order_devices?` are a particular source of confusion. Would you be able to rename them to something longer but hopefully more explicit?

Reply:
Agreed theres quite a few permutations and getting more confusing. I've made it more explicit even though it's not actually needed

Reply:
thanks. I still think that this logic needs an overhaul because:

1. there's a bunch of similar-sounding methods that do similar things and are quite hard to follow
2. there's too much logic in controllers that needs to be pushed down to the model or service level

however that probably shouldn't block this change. I will pick it up separately as I have some ideas.